### PR TITLE
Abort when coroutine is cancelled

### PR DESCRIPTION
**Pull request by rucyang**
There is a scene, while running `async for result in AsyncLLMEngine.generate(...)`，if caller cancel the coroutine at high level (ex. Sanic Framework cancel the coroutine which handling request when transport connection is disconnected), whether to need catching `asyncio.CancelledError ` and aborting  token generating in background?

### CODE REVIEW

File: vllm/engine/async_llm_engine.py [hunk 1]
```diff
@@ -385,8 +385,8 @@ async def generate(
 
             async for request_output in stream:
                 yield request_output
-        except Exception as e:
-            # If there is an exception, abort the request.
+        except (Exception, asyncio.CancelledError) as e:
```
Comment (zhuohan123) [inline vllm/engine/async_llm_engine.py:388]: Why is this needed? I believe `Exception` should already capture all exceptions?

Comment (Yard1) [reply vllm/engine/async_llm_engine.py:388]: > Changed in version 3.8: [CancelledError](https://docs.python.org/3/library/asyncio-exceptions.html#asyncio.CancelledError) is now a subclass of [BaseException](https://docs.python.org/3/library/exceptions.html#BaseException) rather than [Exception](https://docs.python.org/3/library/exceptions.html#Exception).

Apparently not, guess I tripped on that too

**Review by Yard1 — APPROVED** [review]
lgtm

**Review by zhuohan123 — APPROVED** [review]
LGTM! Thanks for the fix!
